### PR TITLE
Summary collector

### PR DESCRIPTION
**Pull request by ZolotarevIgor-mail-ru**
Linked to #97

Ported from: https://github.com/bmizerany/perks/tree/master/quantile

Based on: https://ieeexplore.ieee.org/document/1410103 (there is small differences between this paper and https://www.cs.rutgers.edu/~muthu/bquant.pdf, e.g. invariant for targeted quantiles and inserting ±inf samples)

Performance: 
- Random unsorted data between 0 and 100, 10^8 samples: about 50 seconds in average vs 120 in #100.
- Sorted sequence, 10^8 samples: 20 seconds vs ~100 in #100

### CODE REVIEW

**Comment by ZolotarevIgor-mail-ru** [comment]
Testing results:
 - Summary validates by promtool
 - Works with prometheus
 - Works with influx
Снимок экрана 2020-08-28 в 21 47 58
Снимок экрана 2020-08-28 в 20 51 56
Снимок экрана 2020-08-28 в 20 52 34
Снимок экрана 2020-08-28 в 21 43 40

File: doc/monitoring/api_reference.rst [hunk 1]
```diff
@@ -157,26 +157,54 @@ Histogram
...
 
-Can be used only as a collector for HTTP statistics (described below)
-and cannot be built explicitly.
+.. module:: metrics
 
-.. class:: histogram_obj
+.. function:: summary(name [, help, objectives])
 
-    .. method: collect()
+    Registers a new summary. Quantile computation is based on algorithm `"Effective computation of biased quantiles over data streams" <https://ieeexplore.ieee.org/document/1410103>`_
```
Comment (lenkis) [inline doc/monitoring/api_reference.rst:170]:
```suggestion
    Registers a new summary. Quantile computation is based on the algorithm `"Effective computation of biased quantiles over data streams" <https://ieeexplore.ieee.org/document/1410103>`_
```

File: doc/monitoring/api_reference.rst [hunk 2]
```diff
@@ -157,26 +157,54 @@ Histogram
...
+
+    :param string   name: Collector name. Must be unique.
+    :param string   help: Help description.
+    :param table objectives: Quantiles to observe in form ``{quantile = error, ... }``.
```
Comment (lenkis) [inline doc/monitoring/api_reference.rst:174]:
```suggestion
    :param table objectives: Quantiles to observe in the form ``{quantile = error, ... }``.
```

File: doc/monitoring/api_reference.rst [hunk 3]
```diff
@@ -157,26 +157,54 @@ Histogram
...
+    :return: Summary object
+
+    :rtype: summary_obj
+
+    .. NOTE::
+
+        The summary is just a set of collectors:
+
+        *  ``name .. "_sum"`` - A counter holding the sum of added observations.
+        *  ``name .. "_count"`` - A counter holding number of added observations.
```
Comment (lenkis) [inline doc/monitoring/api_reference.rst:186]:
```suggestion
        *  ``name .. "_count"`` - A counter holding the number of added observations.
```

**Review by lenkis — CHANGES_REQUESTED** [review]

**Review by lenkis — APPROVED** [review]

**Review by vasiliy-t — APPROVED** [review]

**Pull request merged**